libcodechecker/libhandlers/analyze.py

```diff
@@ -177,6 +184,58 @@ def add_arguments_to_parser(parser):
                                help="File containing argument which will be "
                                     "forwarded verbatim for Clang-Tidy.")
 
+    if is_ctu_capable():
+        ctu_opts = parser.add_argument_group('cross translation unit analysis')
+        ctu_mutex_group = ctu_opts.add_mutually_exclusive_group()
+        ctu_mutex_group.add_argument('--ctu',
+                                     action='store_const',
+                                     const=[True, True],
+                                     dest='ctu_phases',
+                                     default=argparse.SUPPRESS,
+                                     help="Perform cross translation unit "
+                                          "(CTU) analysis (both collect and "
+                                          "analyze phases) using default "
+                                          "<ctu-dir> for temporary output. "
+                                          "At the end of the analysis, the "
+                                          "temporary directory is removed.")
```

Comment:
@gerazo Keeping in line with scan-build is understandable. However, there are some concerns we must take to keep the integrity of CodeChecker, and not turn it into a "scanbuild 2".

A concern about `--ctu-dir` being an argument, and `--on-the-fly` not having `--ctu` prefix (thus allowing a collision with other non-CTU option of the same optstring, if such arises), is a problem. I'll comment about `--ctu-dir` below.